bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Control.java

```diff
@@ -3606,6 +3614,23 @@ boolean setRadioSelection (boolean value) {
 	return false;
 }
 
+private boolean hasEmbeddedChildren() {
+	for (Control ec : embeddedControls) {
+		if (!ec.isDisposed() && ec.isVisible() && ec.isChildOf(this)) return true;
+	}
+	return false;
+}
```

Comment:
Can't we calculate this on-the-fly (via the children) instead of managing a data structure storing those widgets?

Reply:
Done. I had to move the check to `Composite` since `Control` doesn't know its children.

Reply:
Good point. How would you suggest to do it?

Reply:
I would simply evaluate this in the `setRedraw()` method of `Control` instead of the one of `Composite`.

Reply:
How would you get the children? Can you provide a snippet?

Reply:
I am afraid I do not understand the question, as I would do it like it is done now, just at a different place. Best may be to use a polymorphic implementation, i.e, something like:

In `Control`:
```java
boolean hasEmbeddedChildren() {
	if (isDisposed() || !isVisible()) {
		return false;
	}
	return (this.style & SWT.EMBEDDED) != 0;
}
```

In `Composite`:
```java
boolean hasEmbeddedChildren() {
	if (super.hasEmbeddedChildren()) {
		return true;
	}
        return Stream.of(getChildren()).anyMatch(Control::hasEmbeddedChildren);
}
```

Reply:
Got it. I added your snippets (with some slight modifications)